src/ServiceControl.Monitoring/Metrics/Raw/LongValueOccurenceSerializerDefinition.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace ServiceControl.Monitoring.Metrics.Raw
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using NServiceBus.MessageInterfaces;
+    using NServiceBus.Serialization;
+    using NServiceBus.Settings;
+
+    class LongValueOccurrenceSerializerDefinition : SerializationDefinition
+    {
+        public override Func<IMessageMapper, IMessageSerializer> Configure(ReadOnlySettings settings)
+        {
+            return mapper => new DurationRawDataSerializer();
+        }
+    }
+
+    class DurationRawDataSerializer : IMessageSerializer
+    {
+        public void Serialize(object message, Stream stream)
+        {
+            throw new NotImplementedException();
+        }
+
+        public object[] Deserialize(Stream stream, IList<Type> messageTypes = null)
+        {
+            var message = new LongValueOccurrences();
+
+            ToLongValueOccurrences(message, stream);
+
+            return new object[]
+            {
+                message
+            };
+        }
+
+        static void ToLongValueOccurrences(LongValueOccurrences message, Stream stream)
+        {
+            var reader = new BinaryReader(stream);
+
+            message.Version = reader.ReadInt64();
```

Comment:
Allocating arrays for ticks and values could lead to a lot of overhead. If we want to use `IMessageSerializer` we could create a pool of `LongValueOccurences` that enables leasing messages and passing them further with no allocations.
An interface could be like:

```
interface IPool 
{
   LongValueOccurrences  Lease (int minCountOfEntries);
   void Return (LongValueOccurences messagE);
}
```

with `Lease` used in the deserialization and `Release` at the end of the handler.